File: lib/calatrava/templates/droid/droid/src/com/calatrava/bridge/KernelBridge.java
package com.calatrava.bridge;

import java.io.IOException;

public class KernelBridge {

  private AssetRepository assetRepository;
  private RhinoService rhinoService;

  public KernelBridge(AssetRepository assetRepository, RhinoService rhinoService) throws IOException {
    this.assetRepository = assetRepository;
    this.rhinoService = rhinoService;

    //load js libraries
    loadLibrary("hybrid/scripts/underscore.js");
    loadLibrary("hybrid/scripts/date.js");

    //load bridge
    loadLibrary("hybrid/scripts/env.js");
    loadLibrary("hybrid/scripts/bridge.js");
  }

  public void loadLibrary(String libraryName) throws IOException {
    rhinoService.load(assetRepository.assetReader(libraryName), libraryName);
  }
}

File: calatrava-ios/calatrava-ios/Bridge/KernelBridge.m
#import "KernelBridge.h"
#include "JSCocoaController.h"
#include "JavascriptCore-dlsym.h"
#include "TWBridgePageRegistry.h"
#include "TWBridgeURLRequestManager.h"

@implementation KernelBridge

+ (void)startWith:(UINavigationController *)root
{
  // Fetch JS symbols
  [JSCocoaSymbolFetcher populateJavascriptCoreSymbols];
    
  // Load iPhone bridgeSupport
  NSString *bundle = [[NSBundle mainBundle] bundlePath];
//  [[BridgeSupportController sharedController] loadBridgeSupport:[NSString stringWithFormat:@"%@/iPhone.bridgesupport",  bundle]];
  id c = [JSCocoaController sharedController];
  [c setUseJSLint:NO];
  // Load js libraries
  [c evalJSFile:[NSString stringWithFormat:@"%@/public/assets/scripts/underscore.js", bundle]];
    
  // Load js bridge
  [c evalJSFile:[NSString stringWithFormat:@"%@/public/assets/scripts/env.js", bundle]];
  [c evalJSFile:[NSString stringWithFormat:@"%@/public/assets/scripts/bridge.js", bundle]];
  NSString *loadFileText = [NSString stringWithContentsOfFile:[NSString stringWithFormat:@"%@/public/assets/load_file.text", bundle]
                                                     encoding:NSASCIIStringEncoding
                                                        error:nil];
  NSArray *jsFiles = [loadFileText componentsSeparatedByString:@"\n"];
  for (NSString *jsFile in jsFiles) {
    [c evalJSFile:[NSString stringWithFormat:jsFile, bundle]];
  }
    
  [[TWBridgePageRegistry sharedRegistry] setRoot:root];
  [[TWBridgeURLRequestManager sharedManager] setRoot:root];
}

+ (void)launch:(NSString *)flow
{
  NSString *bundle = [[NSBundle mainBundle] bundlePath];
  id c = [JSCocoaController sharedController];
  [c evalJSString:flow];
}

@end
